eng/tools/TypeSpecValidation/src/rules/compile.ts

```diff
@@ -17,6 +17,13 @@ export class CompileRule implements Rule {
         `npx --no tsp compile . --warn-as-error`,
         folder,
       );
+      if (
```

Comment:
This feels quite fragile. I think it might be good to consider calling the compiler programmatically to get that info reliabilty.

Reply:
Do you have an example of how we would call the compiler programmatically instead?  I'm not too worried about the fragility of this specific check, since it should just be defense-in-depth for the `emit-autorest` rule.

Reply:
https://github.com/microsoft/typespec/blob/be937a8b850e945073f8711232c73cc5fc604076/packages/samples/src/sample-snapshot-testing.ts#L89-L111

Reply:
https://github.com/Azure/azure-rest-api-specs/issues/26418